Sources/SwiftSMTP/Mail.swift

```diff
@@ -122,27 +123,72 @@ public struct Mail {
         dictionary["MESSAGE-ID"] = id
         dictionary["DATE"] = Date().smtpFormatted
         dictionary["FROM"] = from.mime
-        dictionary["TO"] = to.map { $0.mime }.joined(separator: ", ")
+        dictionary["TO"] = foldHeaderValue(key: "TO", value: to.map { $0.mime }.joined(separator: ", "))
 
         if !cc.isEmpty {
-            dictionary["CC"] = cc.map { $0.mime }.joined(separator: ", ")
+            dictionary["CC"] = foldHeaderValue(key: "CC", value: cc.map { $0.mime }.joined(separator: ", "))
```

Comment:
I agree. The fact, though, is that the first line of the header has to include the key for length calculation. Moreover, I added logging to the folding function so that if someone adds a header that exceeds the maximum length and thus generates a message that an SMTP server would reject, they'd at least have some clue as to which header is at fault. If I were doing this from scratch, I'd probably rewrite the headers not as a dictionary of name: value but as an array of field structs, where each field has a name property and a value property, and the folding logic is encapsulated inside the field.

To my mind, that seems a bit drastic just to address what is ultimately an edge case.

Reply:
I agree, that's probably the better approach.  Also agree it's a big change for what we're trying to fix.